crates/primitives/src/transaction/signature.rs

```diff
@@ -202,6 +202,16 @@ impl Signature {
     pub const fn size(&self) -> usize {
         core::mem::size_of::<Self>()
     }
+
+    /// Converts the signature to an alloy signature.
+    pub fn to_alloy_signature(&self) -> alloy_primitives::Signature {
+        alloy_primitives::Signature::from_rs_and_parity(
+            self.r,
+            self.s,
+            alloy_primitives::Parity::Parity(self.odd_y_parity),
+        )
+        .expect("failed to create alloy signature")
+    }
```

Comment:
@mattsse I'm not super happy with this function but unfortunately we need it to transform our reth `Signature` to an alloy `Signature` to ensure compatibility with the alloy functions of `TxEip7702`...

`from_rs_and_parity` can return an error. If we handle the error in the function then we will have to use `unwrap` everywhere else in the codebase because the functions using this method are not made to fail.

We had discussed this at Frontiers but maybe we should think about unifying the signatures and putting them in alloy to avoid this type of duplication, right?

Reply:
was gonna say seems worth unifying the sigs yea

Reply:
We've just merged #9593 which added `SignatureWithParity` implementing `alloy::EncodableSignature`. #10262 already uses it for endoding 1559 transactions.

The issue with signature from alloy-primitives is that it is impossible to instantiate without validity checks in `k256` crate. I believe this might cause issues for deposit transaction signatures, and in general reth code handles invalid signatures differently.

I think at some point we'll want to change `Signature` type from primitives to make it possible to instantiate invalid signatures and check their validity during recovery, but for now we can just use `SignatureWithParity` which takes advantage of `EncodableSignature` trait to keep vrs encoding logic in alloy

Reply:
Thanks, yes I saw that and I adapted all the PRs accordingly. But indeed, I think it would be a good idea to think about how to unify the signatures because I have the impression that there are too many different types for the same thing between reth and alloy...

We could surely think about a modular primitive type to integrate into alloy and that we could reuse everywhere.